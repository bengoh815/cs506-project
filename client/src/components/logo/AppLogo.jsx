/******************************************************************************
 * Filename: AppLogo.jsx
 * Purpose: A component that displays the Melody Mapper logo.
 * Author:   Victor Nguyen
 *
 * Description:
 * This file contains the AppLogo component that displays the Melody Mapper logo.
 *
 * Usage:
 * To use this component, simply import it into the desired file and render it.
 *
 ******************************************************************************/

import logo from "../../images/melody_logo.png";

export default function AppLogo() {
  return (
    <div id="logo-container">
      <img
        id="logo"
        data-testid="melody-logo"
        className="melody_logo"
        src={logo}
        alt="Melody Mapper Logo"
        height="80"
        style={{ margin: "1rem" }}
      />
    </div>
  );
}
