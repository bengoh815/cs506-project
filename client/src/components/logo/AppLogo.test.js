import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import AppLogo from "./AppLogo";

test("renders AppLogo component", () => {
  // Arrange
  render(<AppLogo />);
  const logoElement = screen.getByTestId("melody-logo");

  // Assert
  expect(logoElement).toBeInTheDocument();
});

// Snapshot test
test("matches snapshot", () => {
  // Arrange
  const tree = renderer.create(<AppLogo />).toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
