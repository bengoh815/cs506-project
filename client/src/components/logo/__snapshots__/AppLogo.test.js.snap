// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="logo-container"
>
  <img
    alt="Melody Mapper Logo"
    className="melody_logo"
    data-testid="melody-logo"
    height="80"
    id="logo"
    src="melody_logo.png"
    style={
      Object {
        "margin": "1rem",
      }
    }
  />
</div>
`;
